lib/hdclient.js

```diff
@@ -3,19 +3,54 @@
 const http = require('http');
 const werelogs = require('werelogs');
 
+const config = require('./config');
+const keyscheme = require('./keyscheme');
+const httpGET = require('./http_get');
+const httpPUT = require('./http_put');
+const httpDELETE = require('./http_delete');
+
 
 class HyperdriveClient {
     /**
      * This represent our interface over several Hyperdrive servers
      * @constructor
      * @param {Object} [opts] - Contains the basic configuration.
-     * @param {werelogs.API} [opts.logApi] - object providing a constructor
+     * @param {werelogs.API} [opts.logApi] - Object providing a constructor
      *                                      function for the Logger object
+     * @param {Object} [opts.policy] - Data placement policy
+     * @param {Number} [opts.dataParts] - Number of data parts
+     * @param {Number} [opts.codingParts] - Number of coding parts
+     * @param {Number} [opts.requestTimeoutMs] - Timeout on answer in ms
+     * @comment For N replication, use data_parts=1 and
+     * coding_parts=N (N+1 part in total)
+     *
+     * @throw config.InvalidConfigError for bad inputs
      */
     constructor(opts) {
-        const options = opts || {};
-        this.httpAgent = new http.Agent({ keepAlive: true });
-        this.setupLogging(options.logApi);
+        this.options = opts;
+
+        const { configIsValid, configError } = config.validate(opts);
+        if (!configIsValid) {
+            throw configError;
+        }
+
+        /* Set backend type as 'scality'
+         * Ths will indicate S3/CloudServer to treat it
+         * specifically (e.g. as SproxydClient)
+         *
+         * DO NOT CHANGE OR 'GET' IS BROKEN
+         */
+        this.clientType = 'scality';
+
+        this.httpAgent = new http.Agent({
+            keepAlive: true,
+            keepAliveMsecs: 30000, // 30s
+            /* Max. number of opened connections towards same hyperdrive */
+            maxSockets: 500,
+        });
+        this.setupLogging(opts.logApi);
+
+        this.setupErrorAgent(opts.errorHandler);
```

Comment:
Indeed. Plus in this form it is misleading. We don't pass the errorHandler itself only potentially some options. Will rework this part. Note an actual, production-ready implementation requires talking to Kafka or Backbeat. Since I can't easily test that today, I preferred to leave it as is, tests with mock and actually plug in the real handler in a separate issue.